client.go

```diff
@@ -929,13 +929,12 @@ func (client *client) tryRefreshMetadata(topics []string, attemptsRemaining int,
 
 // if no fatal error, returns a list of topics that need retrying due to ErrLeaderNotAvailable
 func (client *client) updateMetadata(data *MetadataResponse, allKnownMetaData bool) (retry bool, err error) {
```

Comment:
Do we not want to check for if a client is closed here?

Reply:
yes, you are right